### PR TITLE
TechRadar - urlTarget new attribute

### PR DESCRIPTION
**Pull request by brunopenso**
## Hey, I just made a Pull Request!

Add the urlTarget support by w3c definition and now it is possible to open links on a new window or tab

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Related to issue #8347

### CODE REVIEW

File: plugins/tech-radar/src/utils/components.tsx [hunk 1]
```diff
@@ -14,22 +14,29 @@
...
 };
 
 export const WithLink = ({
   url,
+  urlTarget,
   className,
   children,
 }: WithLinkProps): JSX.Element =>
-  url ? (
-    <a href={url} className={className}>
```
Comment (jhaals) [inline plugins/tech-radar/src/utils/components.tsx:30]: Hmm. I'd actually vote for skipping the new field and passing of target entirely and just make target blank the default unless there are good reasons for not making that the default.

Comment (freben) [reply plugins/tech-radar/src/utils/components.tsx:30]: I'd say don't use target at all. Let the Link decide what to do. It decides what to do based on whether it's an internal or external link, and in the future the Link may even be tied to per user preferences on what preferred target behavior they have. Just leave it unset :)

Comment (jhaals) [reply plugins/tech-radar/src/utils/components.tsx:30]: Even better!

Comment (brunopenso) [reply plugins/tech-radar/src/utils/components.tsx:30]: Let me do some testing!!!

**Review by freben — APPROVED** [review]
Fixed up using Link and removing the new attribute - merging. Thanks!

**Pull request merged**